### ISSUE TITLE
Format versions document

### ISSUE THREAD

**Issue by SirLich** [SirLich/technical-bedrock#7]
This should go in the Knowledge section. 

It should explain, in plain english, what format versions are, and generally how to use them. For example, mention that the Entity format version matches the game version, but that other format versions don't. 

Then, try to compile a table/list of known format options for each category. i.e, all valid animation versions, and the differences between them.

**Comment by SirLich** [comment]
The version is how the client knows if it should retireve a new version, or use the cached version

**Comment by Kas-tle** [comment]
SirLich in Discord:

> A few things I would add:
>  - how to "reason" about format version (mostly suggesting checking vanilla resources/behaviors for inspiration about the latest format version)
>  - tables of format versions for each file-type.
> It should solve the issue of someobdy posting in Bedrock Addons with a format version 1.8 BP entity
> When we say "you need to use modern format version" they say "whats that"?
> Then when we convince them to change it, they ask "what do I change it to?".
> THEN when they change it, they are confused why their entity no longer works (they probably copy/pasted a legacy entity).
> So somehow communicating WHAT format versions do, HOW to understand them, and then WHICH format version to use for any given file.
> But the work is endless... right now I'm in a quality > quantity mindset, so I would rather see existing pages polished rather than new pages introduced.
